maptivist/src/main/java/com/google/sps/servlets/MarkerServlet.java

```diff
@@ -61,23 +67,28 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
         UUID id = UUID.randomUUID();
         Set<String> linkSet = new HashSet<String>(Arrays.asList(request.getParameterValues("marker-links")));
         Set<String> categorySet = new HashSet<String>(Arrays.asList(request.getParameterValues("marker-category")));
+        String dateString = request.getParameter("marker-date");
+
+        Marker postMarker = new Marker(title, description, address, latitude, longitude, linkSet, categorySet, id, dateString);
 
-        Marker postMarker = new Marker(title, description, address, latitude, longitude, linkSet, categorySet, id);
         DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
  
-        // if (!checkIfMarkerAlreadyInDatastore(postMarker.getUUID())) {
+        if (!checkIfMarkerAlreadyInDatastore(postMarker.getUUID())) {
             Entity markerEntity = postMarker.toEntity();
             datastore.put(markerEntity);
-        // }
+        }
  
         response.sendRedirect("/index.html");
     }
  
     private ArrayList<Marker> getMarkers(HttpServletRequest request){
         ArrayList<Marker> markers = new ArrayList<>();
- 
+        LocalDate localDate = LocalDate.now();
+        Date currentDate = localDateToDate(localDate);
+
         DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-        Query query = new Query("Marker");
+        Query query = new Query("Marker")
+        .setFilter(new FilterPredicate("date", FilterOperator.GREATER_THAN_OR_EQUAL	, currentDate.getTime()));
```

Comment:
This is a case where a more precise name than "date" would be useful for code readers. Because it isn't obvious why passing in the currentDate in milliseconds is the right thing to do from the name.

So if this said, "epoch-days" and you passed in "localDate.toEpochDays()" then the code looks obviously correct.